### PR TITLE
feat(governance): DAO proposal for protocol upgrade to Unlock v13 and PublicLock v14

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This PR contains the (cross-chain) DAO protocol for the protocol upgrade to Unlock v12 > v13 and PublicLock v13 > v14

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by clemsos** [comment]
need to rebase when #13203 is merged

**Comment by clemsos** [comment]
Upgrade protocol: switch to Unlock v13 and PublicLock v14

### Goal of the proposal

This is a proposal for an upgrade of the core Unlock Protocol smart contracts - with the new Unlock Version (v13) and PublicLock version (v14).

# About the upgrade

This upgrade includes improved mechanics for UDT token governance, several improvements in existing features, gas optimisations and bug fixes. The main novelty is the new “swap and burn” feature that will allow fees collected by the protocol to be directly decrease the supply of UDT in circulation.

# A cross-chain proposal

The proposal uses a cross-chain proposal pattern that, once passed, will deploy the upgrade on multiple chains at once. This pattern has been introduced and tested in a [previous proposal](https://www.tally.xyz/gov/unlock/proposal/1926572528290918174819693611122933562560576845671089759587616947457423587439).

# The calls

All calls are sent to the Connext bridge at 0x8898B472C54c31894e3B9bb83cEA802a5d0e63C6 on chain 1:



### Optimism (chain 10)

- `addLockTemplate(0x530Ff2dAED410cA7D70C25f18dc770f106201151,14)` - Unlock at 0x99b1348a9129ac49c6de7F11245773dE2f51fB0c
- `setLockTemplate(0x530Ff2dAED410cA7D70C25f18dc770f106201151)` - Unlock at 0x99b1348a9129ac49c6de7F11245773dE2f51fB0c
- `upgrade(0x99b1348a9129ac49c6de7F11245773dE2f51fB0c,0x508619074f542b6544c5835f260CC704E988cf65)` - ProxyAdmin at 0xA2D9BC1ffc560Cc6F21aC09E317A3186AC1B1db8


### BNB Chain (chain 56)

- `addLockTemplate(0xA8BB5AF09B599794136B14B112e137FAf83Acf1f,14)` - Unlock at 0xeC83410DbC48C7797D2f2AFe624881674c65c856
- `setLockTemplate(0xA8BB5AF09B599794136B14B112e137FAf83Acf1f)` - Unlock at 0xeC83410DbC48C7797D2f2AFe624881674c65c856
- `upgrade(0xeC83410DbC48C7797D2f2AFe624881674c65c856,0xfe9fD6af67E48D9f05Aa88679Ac294E3f28532eE)` - ProxyAdmin at 0xA2D9BC1ffc560Cc6F21aC09E317A3186AC1B1db8


### Gnosis Chain (chain 100)

- `addLockTemplate(0xeAd6d1877452383ab5F74c689b6C3d0538Fd3008,14)` - Unlock at 0x1bc53f4303c711cc693F6Ec3477B83703DcB317f
- `setLockTemplate(0xeAd6d1877452383ab5F74c689b6C3d0538Fd3008)` - Unlock at 0x1bc53f4303c711cc693F6Ec3477B83703DcB317f
- `upgrade(0x1bc53f4303c711cc693F6Ec3477B83703DcB317f,0x24BF5517Ecc83caB64478Ab3D69950aA1567eB89)` - ProxyAdmin at 0xA2D9BC1ffc560Cc6F21aC09E317A3186AC1B1db8


### Polygon (chain 137)

- `addLockTemplate(0x8231d6fD0221C01FCAc5827EdD20D1aeC28EeBe3,14)` - Unlock at 0xE8E5cd156f89F7bdB267EabD5C43Af3d5AF2A78f
- `setLockTemplate(0x8231d6fD0221C01FCAc5827EdD20D1aeC28EeBe3)` - Unlock at 0xE8E5cd156f89F7bdB267EabD5C43Af3d5AF2A78f
- `upgrade(0xE8E5cd156f89F7bdB267EabD5C43Af3d5AF2A78f,0x4132f269168375DBf7DcDb2cfEA348F453FD4B40)` - ProxyAdmin at 0xA2D9BC1ffc560Cc6F21aC09E317A3186AC1B1db8

Onwards !

The Unlock Protocol Team

File: governance/proposals/009-protocol-upgrade.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+  // set proposal name
+  const proposalName = `Upgrade protocol: switch to Unlock v13 and PublicLock v14
+  
+## Goal of the proposal
+
+This is a proposal for an upgrade of the core Unlock Protocol smart contracts - with the new Unlock Version (v13) and PublicLock version (v14). 
+
+## About the upgrade
+
+This upgrade includes improved mechanics for UDT token governance, several improvements in existing features, gas optimisations and bug fixes. The main novelty is the new “swap and burn” feature that will allow fees collected by the protocol to be directly decrease the supply of UDT in circulation.
```
Comment (julien51) [inline governance/proposals/009-protocol-upgrade.js:247]:
```suggestion
This upgrade includes improved mechanics for UDT token governance, several improvements in existing features, gas optimisations and bug fixes. The main novelty is the new “swap and burn” feature that will allow fees collected by the protocol to directly decrease the supply of UDT in circulation.
```

**Comment by ccarfi** [comment]
made one small copy change in the About comment above. for the changes in the 15 js and json files themselves, need to defer to you all.

**Comment by clemsos** [comment]
live at https://www.tally.xyz/gov/unlock/proposal/67650238492744053870551188976528991701570154819711289520259087104693406921512?chart=2 !

**Pull request merged**